### PR TITLE
Pass docid to choices fns

### PR DESCRIPTION
**Pull request by stuartromanek**
## Summary

- Passes parent `docId` to dynamic choice functions in schemas (selects, checkboxes, radios) for further inspections
- Properly passes `docId` through array fields and into their child schemas 

## What kind of change does this PR introduce?
*(Check at least one)*

- [X] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [X] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: modules/@apostrophecms/schema/index.js [hunk 1]
```diff
@@ -1484,7 +1485,7 @@ module.exports = {
           ) {
             throw self.apos.error('invalid');
           }
-          choices = await self.apos.modules[field.moduleName][field.choices](req);
+          choices = await self.apos.modules[field.moduleName][field.choices](req, docId);
```
Comment (boutell) [inline modules/@apostrophecms/schema/index.js:1488]: Please put that in an object, I'm sure there will be more.

File: modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue [hunk 2]
```diff
@@ -109,6 +110,10 @@ export default {
     serverError: {
       type: Object,
       default: null
+    },
+    docId: {
```
Comment (boutell) [inline modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue:114]: Thanks. Is it already passed to object fields?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,10 +4,12 @@
 
 ### Adds
 
+* Dynamic choice functions in schemas now recieve their original doc id for further inspection by your function.
```
Comment (boutell) [inline CHANGELOG.md:7]: Make sure you request a docs update from Bob (or just PR one).

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/schema/index.js (hunk 1 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/index.js]: i knew this!!!

File: modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue (hunk 2 above)
Comment (stuartromanek) [reply modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue:114]: updated to pass it through and confirmed.

File: CHANGELOG.md (hunk 3 above)
Comment (stuartromanek) [reply CHANGELOG.md]: yep he's got one

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,10 +4,12 @@
 
 ### Adds
 
+* Dynamic choice functions in schemas now als recieve a data object with their original doc id for further inspection by your function.
```
Comment (boutell) [inline CHANGELOG.md:7]: als -> also (sorry)

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,10 +4,12 @@
...
 * Use `mergeWithCustomize` when merging extended source Webpack configuration. Introduce overideable asset module methods `srcCustomizeArray` and `srcCustomizeObject`, with reasonable default behavior, for fine tuning Webpack config arrays and objects merging. More info - [the Webpack mergeWithCustomize docs](https://github.com/survivejs/webpack-merge#mergewithcustomize-customizearray-customizeobject-configuration--configuration)
 
 ### Fixes
 
+* `docId` is now properly passed through array fields and into it's child schemas.
```
Comment (boutell) [inline CHANGELOG.md:12]: it's -> its

Comment (boutell) [reply CHANGELOG.md:12]: it's -> their would be best

File: modules/@apostrophecms/schema/index.js [hunk 6]
```diff
@@ -1484,7 +1485,7 @@ module.exports = {
           ) {
             throw self.apos.error('invalid');
           }
-          choices = await self.apos.modules[field.moduleName][field.choices](req);
+          choices = await self.apos.modules[field.moduleName][field.choices](req, { docId });
```
Comment (boutell) [inline modules/@apostrophecms/schema/index.js:1488]: 💯

**Review by boutell — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (stuartromanek) [reply CHANGELOG.md]: updated

**Review by boutell — APPROVED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**